### PR TITLE
Added subtitle support. Fixes#35

### LINKED ISSUE #35: srt file support without transcoding
Does dms support srt file over dlna without transcoding ?
I tried dms on my rpi 2 but its not sending subtitle to my LG tv.
Video works fine.

### PR DESCRIPTION
**Pull request by abdullah2993**
Added support for subtitle, It works but  this might not be the most elegant solution. IMO the best approach should be to find subtitle files based on filename when walking the FS and then only add the subtitle res for objects for which the subtitles were found.
Fixes #35

### CODE REVIEW

File: dlna/dms/dms.go [hunk 1]
```diff
@@ -722,6 +729,7 @@ func (server *Server) initMux(mux *http.ServeMux) {
 	})
 	mux.HandleFunc(contentDirectoryEventSubURL, server.contentDirectoryEventSubHandler)
 	mux.HandleFunc(iconPath, server.serveIcon)
+	mux.HandleFunc(subtitlePath, server.serveIcon)
```
Comment (anacrolix) [inline dlna/dms/dms.go]: `server.serveIcon`?

Comment (abdullah2993) [reply dlna/dms/dms.go]: Sorry messed up the commit, Fixed

**Comment by anacrolix** [comment]
Thanks, I've mentioned it in the linked issue to get hopefully get some outside input.

**Comment by halfwit** [comment]
You could crawl the directory that the video is in, for a file with a conforming mimetype; we have mimesniff support in the http package. Personally I just remuxed my files to include the subtitles, but it would be great to not have to

**Comment by jakima** [comment]
I've downloaded the repo, made the suggested changes in this PR and built it locally on Windows.
Ran it on a folder, that has some movies with external subs and the other are mkvs.

First test was my phone running just plain VLC. It worked flawlessly. External subs were served(?) with their respective videos.

Second test was my "smart" TV running bubbleupnp as VLC has [a bug/oversight](https://code.videolan.org/videolan/vlc-android/-/issues/2187) that it doesn't start scanning unless the device running VLC is explicitly connected to a network instead of being in a network (such as opening a hotspot as is the case with my TV).
Sadly, bubbleupnp couldn't pick up the subtitle file.
Connecting laptop to TV's hotspot because, privacy concerns. Stock OS in TV has lots of things that call home.

I happen to find another program called [nano-dlna](https://github.com/gabrielmagno/nano-dlna/). It seems to have no problems serving a subtitle file that both VLC and bubbleupnp picked up. Although, in its current state, only a single file could be served.
But, it has an [xml template](https://github.com/gabrielmagno/nano-dlna/blob/master/nanodlna/templates/metadata-video_subtitle.xml) just for subtitle. I think this might be where the problem lies. Though, I could be wrong.

That said, great work. I like this project.

**Comment by anacrolix** [comment]
Thank you @jakima. This project likes you too. I think I'll merge this PR shortly based on your feedback. It's also a very tidy PR.

**Comment by anacrolix** [comment]
Thanks @jakima and @abdullah2993 .

**Comment by enihcam** [comment]
I have a domestic-brand TV with a basic DLNA player app by which the subtitles (.srt) file cannot be scanned/detected, even though i renamed the srt filename to exactly the video file name. any suggestions? thanks.

**Comment by anacrolix** [comment]
It's been a while but the first thing that comes to mind is to check the mime type for .srt files on your system.

**Comment by enihcam** [comment]
> It's been a while but the first thing that comes to mind is to check the mime type for .srt files on your system.

thanks but where to config the mime type of .srt?

**Comment by abdullah2993** [comment]
> It's been a while but the first thing that comes to mind is to check the mime type for .srt files on your system.

Also do try to play the same using vlc on a computer or using mx player on your android phone. Just to rule out any issues related to naming.

**Pull request merged**